shell/common/gin_converters/gfx_converter.cc

```diff
@@ -230,4 +231,352 @@ bool Converter<WrappedSkColor>::FromV8(v8::Isolate* isolate,
   return true;
 }
 
+v8::Local<v8::Value> Converter<gfx::ColorSpace>::ToV8(
+    v8::Isolate* isolate,
+    const gfx::ColorSpace& val) {
+  auto dict = gin_helper::Dictionary::CreateEmpty(isolate);
+
+  // Convert primaries to string
+  std::string primaries;
+  switch (val.GetPrimaryID()) {
+    case gfx::ColorSpace::PrimaryID::BT709:
+      primaries = "bt709";
+      break;
+    case gfx::ColorSpace::PrimaryID::BT470M:
+      primaries = "bt470m";
+      break;
+    case gfx::ColorSpace::PrimaryID::BT470BG:
+      primaries = "bt470bg";
+      break;
+    case gfx::ColorSpace::PrimaryID::SMPTE170M:
+      primaries = "smpte170m";
+      break;
+    case gfx::ColorSpace::PrimaryID::SMPTE240M:
+      primaries = "smpte240m";
+      break;
+    case gfx::ColorSpace::PrimaryID::FILM:
+      primaries = "film";
+      break;
+    case gfx::ColorSpace::PrimaryID::BT2020:
+      primaries = "bt2020";
+      break;
+    case gfx::ColorSpace::PrimaryID::SMPTEST428_1:
+      primaries = "smptest428-1";
+      break;
+    case gfx::ColorSpace::PrimaryID::SMPTEST431_2:
+      primaries = "smptest431-2";
+      break;
+    case gfx::ColorSpace::PrimaryID::P3:
+      primaries = "p3";
+      break;
+    case gfx::ColorSpace::PrimaryID::XYZ_D50:
+      primaries = "xyz-d50";
+      break;
+    case gfx::ColorSpace::PrimaryID::ADOBE_RGB:
+      primaries = "adobe-rgb";
+      break;
+    case gfx::ColorSpace::PrimaryID::APPLE_GENERIC_RGB:
+      primaries = "apple-generic-rgb";
+      break;
+    case gfx::ColorSpace::PrimaryID::WIDE_GAMUT_COLOR_SPIN:
+      primaries = "wide-gamut-color-spin";
+      break;
+    case gfx::ColorSpace::PrimaryID::CUSTOM:
+      primaries = "custom";
+      break;
+    case gfx::ColorSpace::PrimaryID::EBU_3213_E:
+      primaries = "ebu-3213-e";
+      break;
+    default:
+      primaries = "invalid";
+      break;
+  }
```

Comment:
When we upgrade Chromium, it may include new enum values upstream. Ideally we want to catch this when upgrading so we can add the new values.

I think if you remove the `default` case, it will fail to compile if there's an unaccounted enum value which would be nice.
https://chromium.googlesource.com/chromium/src/+/main/styleguide/c++/checks.md#:~:text=hear%20about%20it.-,switch%20(my_enum)%20%7B,-case%20A%3A

Reply:
I verified this is the case - if you drop the `default` case, you will get an error like this :
```
../../electron/shell/common/gin_converters/gfx_converter.cc:241:15: error: enumeration values 'INVALID' and 'FILM' not handled in switch [-Werror,-Wswitch]
  241 |   switch (val.GetPrimaryID()) {
```

Reply:
Dropped